txpool/txpool.go

```diff
@@ -823,466 +531,508 @@ func (t *TxPool) validateTx(
-}
 
-func (t *txHeap) Len() int {
-	return len(*t)
+	return account.getNonce()
 }
 
-func (t *txHeap) Swap(i, j int) {
-	(*t)[i], (*t)[j] = (*t)[j], (*t)[i]
+// GetCapacity returns the current number of slots
+// occupied in the pool as well as the max limit
+func (p *TxPool) GetCapacity() (uint64, uint64) {
+	return p.gauge.read(), p.gauge.max
 }
 
-func (t *txHeap) Less(i, j int) bool {
-	return (*t)[i].Nonce < (*t)[j].Nonce
+// GetPendingTx returns the transaction by hash in the TxPool (pending txn) [Thread-safe]
+func (p *TxPool) GetPendingTx(txHash types.Hash) (*types.Transaction, bool) {
+	tx, ok := p.index.load(txHash)
+	if !ok {
+		return nil, false
+	}
+
+	return tx, true
 }
 
-func (t *txHeap) Push(x interface{}) {
-	(*t) = append((*t), x.(*types.Transaction))
+// GetTxs gets pending and queued transactions
+func (p *TxPool) GetTxs(inclQueued bool) (
+	allPromoted, allEnqueued map[types.Address][]*types.Transaction,
+) {
+	allPromoted, allEnqueued = p.accounts.allTxs(inclQueued)
+	return
 }
 
-func (t *txHeap) Pop() interface{} {
-	old := *t
-	n := len(old)
-	x := old[n-1]
-	*t = old[0 : n-1]
-	return x
+// Thread safe map of all accounts registered by the pool.
+type accountsMap struct {
+	sync.Map
+	count uint64
 }
 
-// Price based heap implementation //
-type pricedTx struct {
-	tx    *types.Transaction
-	from  types.Address
-	price *big.Int
-	index int
+// Intializes an account for the given address.
+func (m *accountsMap) initOnce(addr types.Address, nonce uint64) *account {
+	a, _ := m.LoadOrStore(addr, &account{})
+	newAccount := a.(*account)
+
+	// run only once
+	newAccount.init.Do(func() {
+		// create queues
+		newAccount.enqueued = newAccountQueue()
+		newAccount.promoted = newAccountQueue()
+
+		// set the nonce
+		newAccount.setNonce(nonce)
+
+		// update global count
+		atomic.AddUint64(&m.count, 1)
+	})
+
+	return newAccount
 }
 
-// helper object for tx price heap
-type txPriceHeap struct {
-	lock  sync.Mutex
-	index map[types.Hash]*pricedTx
-	heap  heap.Interface
+// Checks if an account exists within the map.
+func (m *accountsMap) exists(addr types.Address) bool {
+	_, ok := m.Load(addr)
+	return ok
 }
 
-func (t *txPriceHeap) Length() uint64 {
-	t.lock.Lock()
-	defer t.lock.Unlock()
+// Collects the primaries of all accounts.
+func (m *accountsMap) getPrimaries() (primaries transactions) {
+	m.Range(func(key, value interface{}) bool {
+		account := m.from(key.(types.Address))
+
+		account.promoted.lock(false)
+		defer account.promoted.unlock()
+
+		// add primary
+		if tx := account.promoted.peek(); tx != nil {
+			primaries = append(primaries, tx)
+		}
+
+		return true
+	})
 
-	return uint64(len(t.index))
+	return primaries
 }
 
-// dropTx removes a transaction from the priced heap. [Thread-unsafe]
-func (t *txPriceHeap) dropTx(tx *types.Transaction) bool {
-	item, ok := t.index[tx.Hash]
+// Returns the account associated with the given address.
+func (m *accountsMap) from(addr types.Address) *account {
```

Comment:
from is a bit misleading, I think renaming to `get` or `find` is more appropriate

Reply:
P.S. you used the verb `load` for the same thing with the lookupMap

Reply:
[Renamed](https://github.com/0xPolygon/polygon-sdk/pull/287/commits/034e1c7dec054cece5fa24dad0deb98f77404b77)